ERCS/erc-7586.md

```diff
@@ -0,0 +1,195 @@
+    */
+    function assetContract() external view returns(address);
+
+    /**
+    *  @notice Returns the notional amount in unit of asset to be transferred when swapping IRS. This amount serves as the basis for calculating the interest payments, and may not be exchanged
+    *          Example: If the two parties aggreed to swap interest rates in USDC, then the notional amount may be equal to 1,000,000 USDC, etc 
+    */
+    function notionalAmount() external view returns(uint256);
+
+    /**
+    *  @notice Returns the interest payment frequency
+    */
+    function paymentFrequency() external view returns(uint256);
+
+    /**
+    *  @notice Returns an array of specific dates on which the interest payments are exchanged. Each date MUST be a Unix timestamp like the one returned by block.timestamp
+    *          The length of the array returned by this function MUST equal the total number of swaps that should be realized
+    *
+    *  OPTIONAL
+    */
+    function paymentDates() external view returns(uint256[] memory);
+
+    /**
+    *  @notice Returns the starting date of the swap contract. This is a Unix Timestamp like the one returned by block.timestamp
+    */
+    function startingDate() external view returns(uint256);
+
+    /**
+    *  @notice Returns the maturity date of the swap contract. This is a Unix Timestamp like the one returned by block.timestamp
+    */
+    function maturityDate() external view returns(uint256);
+
+    /**
+    *  @notice Returns the benchmark in basis point unit
+    *          Example: value of one the following rates: CF BIRC, EURIBOR, HIBOR, SHIBOR, SOFR, SONIA, TONAR, etc.
+    */
+    function benchmark() external view returns(uint256);
+
+    /**
+    *  @notice Returns the oracle contract address for the benchmark rate.
+    *          This contract SHOULD be used to fetch real time benchmark rate
+    *          Example: Contract address for `CF BIRC`
+    *
+    *  OPTIONAL. The two parties MAY agree to set the benchmark manually
+    */
+    function oracleContractForBenchmark() external view returns(address);
+
+    /**
+    * @notice Gives agreement to swap IRS. After the contract deployment and prior to the initiation of the first payment, it is imperative for both the `payer` and the `receiver` to invoke this function as an expression of their agreement
+    */
+    function agree() external;
+
+    /**
+    *  @notice Makes swap calculation and transfers the interest difference to either the `payer` or the `receiver`
+    */
+    function swap() external returns(bool);
+
+    /**
+    *  @notice Terminates the swap contract before its maturity date. It is RECOMMENDED to integrate a multisig process to terminate the swap contract.
+    *          Both the payer and the receiver MUST sign for the termination to be successful
```

Comment:
In a technical sense, what does "sign" mean here?

Reply:
By `sign` I meant they must give their agreement to terminate the contract. This can be done off-chain the same way as you suggested for the `agree` function.